### ISSUE TITLE
Running EITrans across multi nodes

### ISSUE THREAD

**Issue by Weiming-Hu** [Weiming-Hu/EITrans#2]
When I'm trying to run EITrans on multi nodes, I received errors complaining about not enough slots are found while I have requested enough nodes and cores.

**Comment by Weiming-Hu** [comment]
Some potentially helpful posts:

- [Installing R on HLRN (Cray XC30/XC40) and enabling (R)MPI](http://www.ahschulz.de/2015/01/02/installing-r-on-hlrn-cray-xc30xc40-and-enabling-rmpi/)
- [Installing Rmpi with extra arguments](https://stackoverflow.com/questions/13570873/installing-rmpi-on-lam-mpi-cluster)
- [doMPI](https://rdrr.io/cran/doMPI/f/inst/doc/doMPI.pdf)

After reading a bunch of posts, I have figured out the following minimal working version with bare bone MPI.

`test.R`

```
suppressMessages(library(doMPI))

cl = startMPIcluster()
registerDoMPI(cl)

print('This is only done once!')

results <- foreach(index = 1:1000, .combine = rbind) %dopar% {
 Sys.info()[c("nodename","machine")]
}

print(unique(results[, 1]))

closeCluster(cl)
mpi.quit()
```

To run it on compute nodes, use `mpirun R --slave -f test.R`. It will print out the unique nodes used during the process, so it is preferred to request for more than one node.

I have tested the same code with `doSNOW` on multi-nodes but it didn't go through. All the references online about `doSNOW` seem to deal with only multi-core.

So I'll probably need to change my backend from `doSNOW` to `doMPI` specifically to use `MPI`.
